girder_worker/plugins/docker/__init__.py

```diff
@@ -86,17 +88,26 @@ def task_cleanup(e):
     from .executor import DATA_VOLUME
     if e.info['task']['mode'] == 'docker' and '_tempdir' in e.info['kwargs']:
         tmpdir = e.info['kwargs']['_tempdir']
-        cmd = [
-            'docker', 'run', '--rm', '-v', '%s:%s' % (tmpdir, DATA_VOLUME),
-            'busybox', 'chmod', '-R', 'a+rw', DATA_VOLUME
-        ]
-        p = subprocess.Popen(args=cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        out, err = p.communicate()
-        if p.returncode:
-            logger.error(
-                'Error setting perms on docker tempdir %s.\nSTDOUT: %s\nSTDERR:%s',
-                tmpdir, out, err)
-            raise Exception('Docker tempdir chmod returned code %d.' % p.returncode)
+        client = docker.from_env()
+        config = {
+            'tty': True,
+            'volumes': {
+                tmpdir: {
+                    'bind': DATA_VOLUME,
+                    'mode': 'rw'
+                }
+            },
+            'detach': False,
+            'remove': True
+        }
+        args = ['chmod', '-R', 'a+rw', DATA_VOLUME]
+
+        try:
+            client.containers.run('busybox', args, **config)
+        except DockerException as dex:
+            logger.error('Error setting perms on docker tempdir %s.' % tmpdir)
+            logger.error(dex)
```

Comment:
I think `logger.exception()` will automatically log the most recently raised exception.